src/pybind/mgr/volumes/fs/operations/trash.py

```diff
@@ -80,6 +85,67 @@ def rmtree(root_path):
             # (else we would fail to remove this anyway)
             if not should_cancel():
                 self.fs.rmdir(root_path)
+                mpr.inc_count()
+
+        class MeasurePurgeRate:
+
+            def __init__(self, period, purge_queue):
+                # measuring period -- period during which attempt to measure
+                # current purge rate is made
+                self.period = period
+                # this instance variable allows measuring only when measuring
+                # period is on
+                self.measuring = True
+
+                self.count = 0
+                self.rate = 0
+                self.time1 = None
+                self.time2 = None
+
+            def inc_count(self):
+                if not self.measuring:
+                    return
+
+                if self.count == 0:
+                    self.time1 = monotonic()
+                else:
+                    assert self.time1
+                    self.time2 = monotonic()
+                self.count += 1
+
+                if self.time2:
+                    time_diff = self.time2 - self.time1
+                    if time_diff >= self.period:
+                        self.rate = round(self.count / time_diff, 3)
+                        log.debug(f'purge rate = {self.rate}')
+                        # save the "purge rate" in purge queue object so that
+                        # it can be accessed in volume.py
+                        purge_queue.purge_rate = self.rate
```

Comment:
Should the `purge_queue.purge_rate` also be reset ?

Reply:
We already do it. See - https://github.com/ceph/ceph/pull/59713/files#diff-adf1b497bf9b7b60e53497d8a88519f2e693b1fa9bd6854f9b72ac75944c61d1R1179